qcodes/instrument/delegate/delegate_instrument.py

```diff
@@ -264,6 +275,99 @@ def _create_and_add_parameter(
             **kwargs
         )
 
+    def _create_and_add_channels(
+        self,
+        station: Station,
+        channels: Union[Mapping[str, Sequence[str]], Mapping[str, str]],
+    ) -> None:
+        """Add channels to the instrument.
+
+        .. code-block:: yaml
+
+        field:
+            type: qcodes.instrument.delegate.DelegateInstrument
+            init:
+            channels:
+                type: mypackage.mydevice.mydevice_channel.MyDeviceChannel
+                top_barrier:
+                  channel: dac.ch01
+                  my_input_param: 0
+
+        Args:
+            station: QCoDeS station, i.e. representation of the entire physical
+                setup.
+            channels: Mapping of names/aliases to instrument channels and
+                optional input parameters if a channel wrapper class is used to
+                instantiate the channel. If no 'type' field is given, the
+                channel is added as is using ``self.add_submodule``.
+
+        """
+
+        channel_wrapper = None
+        if 'type' in channels.keys():
+            module_name = '.'.join(channels['type'].split('.')[:-1])
+            instr_class_name = channels['type'].split('.')[-1]
+            module = importlib.import_module(module_name)
+            channel_wrapper = getattr(module, instr_class_name)
```

Comment:
I think this can be written a bit cleaner as

```suggestion
        channel_type = channels.pop("type", None)
        if channel_type is not None:
            channel_type_elems = channel_type.split(".")
            module_name = '.'.join(channel_type_elems[:-1])
            instr_class_name = channel_type_elems[-1]
            module = importlib.import_module(module_name)
            channel_wrapper = getattr(module, instr_class_name)
```

Then we can also drop the check for ken != type below